### PR TITLE
Open id connect authentication

### LINKED ISSUE #337: Configure SSO/OAuth integration for API Manager
It would be useful to be able to configure authentication (Authn/Authz) providers from the Operator to fully enable Infrastructure-as-Code. For example, being able to deploy Keycloak and then provision the APIManager to use that KeyCloak instance based on the contents of the CR. Perhaps a new/separate CRD for Auth providers?

### LINKED ISSUE #351: Add support to configure oidc as an authentication mecanism
Add support to configure oidc as an authentication mecanism using API CRD

### PR DESCRIPTION
**Pull request by eguzki**
Fixes #337
Fixes #351

### CODE REVIEW

File: apis/capabilities/v1beta1/product_types.go [hunk 1]
```diff
@@ -322,14 +322,86 @@ func (u *UserKeyAuthenticationSpec) GatewayResponseSpec() *GatewayResponseSpec {
 	return u.GatewayResponse
 }
 
+// OIDCAuthenticationFlowSpec defines the desired OAuth2.0 authorization grant type
+type OIDCAuthenticationFlowSpec struct {
```
Comment (miguelsorianod) [inline apis/capabilities/v1beta1/product_types.go:326]: Are they all optional? if so we should set them as pointers and mark them as optional appropriately as per k8s API conventions guide (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md)

**Review by miguelsorianod — CHANGES_REQUESTED** [review]
Looks good to me.

Just a question about a clarification regarding the optionality of some attributes

File: apis/capabilities/v1beta1/product_types.go (hunk 1 above)
Comment (eguzki) [reply apis/capabilities/v1beta1/product_types.go:326]: They are all required. It is the attribute of the OIDCSpec `AuthenticationFlow` the one which is optional.
```
// AuthenticationFlow specifies OAuth2.0 authorization grant type                              
// +optional                                                                                   
AuthenticationFlow *OIDCAuthenticationFlowSpec `json:"authenticationFlow,omitempty"`           
```

Comment (miguelsorianod) [reply apis/capabilities/v1beta1/product_types.go:326]: :+1:

**Review by miguelsorianod — APPROVED** [review]

**Pull request merged**